src/table/io/FitsReader.cc

```
@@ -1,547 +1,39 @@
-            fits->status = 0;
-            flagCol = -1;
-        }
-        fits->behavior |= Fits::AUTO_CHECK;
-    }
-
-    std::size_t const & row;  // this is a reference back to the _row data member in FitsReader
-    mutable int col;          // the current column (0-indexed)
-    mutable int bit;          // the current flag bit (in the FITS table, not the Schema)
-    int nFlags;               // the total number of flags
-    int flagCol;              // the column number (0-indexed) that holds all the flag bits
-    Fits * fits;              // the FITS file pointer
-    boost::scoped_array<bool> flags;  // space to hold a bool array of the flags to pass to cfitsio
-    BaseRecord * record;      // record to write values to
-};
-
-PTR(BaseRecord) FitsReader::_readRecord(PTR(BaseTable) const & table) {
-    PTR(BaseRecord) record;
-    if (++_row == _nRows) return record;
-    record = table->makeRecord();
-    _processor->record = record.get();
-    _processor->apply(table->getSchema());
-    return record;
-}
-
-// ------------ FitsReader Registry implementation ----------------------------------------------------------
-
-namespace {
-
-typedef std::map<std::string,FitsReader::Factory*> Registry;
+typedef std::map<std::string,FitsReader const*> Registry;
 
 Registry & getRegistry() {
     static Registry it;
     return it;
 }
 
-// here's an example of how you register a FitsReader
-static FitsReader::FactoryT<FitsReader> baseReaderFactory("BASE");
+static FitsReader const baseFitsReader("BASE");
 
 } // anonymous
 
-FitsReader::Factory::Factory(std::string const & name) {
+PTR(BaseTable) FitsReader::makeTable(
+    FitsSchemaInputMapper & mapper,
+    PTR(daf::base::PropertyList) metadata,
+    int ioFlags,
+    bool stripMetadata
+) const {
+    PTR(BaseTable) result = BaseTable::make(mapper.finalize());
+    result->setMetadata(metadata);
+    return result;
+}
+
+FitsReader::FitsReader(std::string const & name) {
     getRegistry()[name] = this;
```

So if you construct multiple ``FitsReader``s with the same name, the last one will take precedence. Is that desirable? Would it be better to throw an error if the name is already taken?